meta/meta_test.go

```diff
@@ -330,248 +323,284 @@ func (s *testSuite) TestElement(c *C) {
 	checkElement(key, errors.Errorf("invalid encoded element key prefix %q", key[:5]))
 
 	_, err := meta.DecodeElement([]byte("_col"))
-	c.Assert(err.Error(), Equals, `invalid encoded element "_col" length 4`)
+	require.EqualError(t, err, `invalid encoded element "_col" length 4`)
 	_, err = meta.DecodeElement(meta.ColumnElementKey)
-	c.Assert(err.Error(), Equals, `invalid encoded element "_col_" length 5`)
+	require.EqualError(t, err, `invalid encoded element "_col_" length 5`)
 }
 
-func (s *testSuite) TestDDL(c *C) {
-	defer testleak.AfterTest(c)()
-	store, err := mockstore.NewMockStore()
-	c.Assert(err, IsNil)
-	defer func() {
-		err := store.Close()
-		c.Assert(err, IsNil)
-	}()
-
-	txn, err := store.Begin()
-	c.Assert(err, IsNil)
+func TestDDL(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		desc        string
+		startHandle kv.Handle
+		endHandle   kv.Handle
+	}{
+		{
+			"kv.IntHandle",
+			kv.IntHandle(1),
+			kv.IntHandle(2),
+		},
+		{
+			"kv.CommonHandle",
+			testkit.MustNewCommonHandle(t, "abc", 1222, "string"),
+			testkit.MustNewCommonHandle(t, "dddd", 1222, "string"),
+		},
+	}
 
-	t := meta.NewMeta(txn)
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			store, err := mockstore.NewMockStore()
+			require.NoError(t, err)
+			defer func() {
+				err := store.Close()
+				require.NoError(t, err)
+			}()
```

Comment:
Maybe share one store if it is not parallel.

Reply:
It seems store state conflict with each other that if I factor out store creation, it failed with

```
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 3
        	Test:       	TestDDL/kv.CommonHandle
```

```go
			job := &model.Job{ID: 1}
			err = m.EnQueueDDLJob(job)
			require.NoError(t, err)
			n, err := m.DDLJobQueueLen()
			require.NoError(t, err)
			require.Equal(t, int64(1), n) // !FAIL!
```

does `store` has a method to clear state?

Reply:
otherwise I suspect whether we run in cases, the original case look fall through and not quite repeat itself.

Reply:
https://github.com/pingcap/tidb/pull/26882/commits/d39ed2d2131cc31f6eebea3df4b0e173028b1cb3 this commit seems not migrate test equivalently . cc @yedamao

Reply:
could you give more details? I don't got it

Reply:
@xhebox @tisonkun  
1. the original TestDDL will rerun by `util/testutil.CommonHandleSuite.RerunWithCommonHandleEnabled`,actually, it will run twice, and the store will also be created twice. this is equivalent with the new version. 
2. store can't be shared between sub tests, because it has state can't be cleanup.

Reply:
> does store has a method to clear state?

No one line solution.

> actually, it will run twice, and the store will also be created twice. this is equivalent with the new version.

You are right, then maybe parallel these two sub tests rather than `TestDDL`. They are distinct.

Reply:
@xhebox I think I get your point. Given it is a common case we run in two handles, I'd prefer introducing a new testkit to replace the original testutil.

Reply:
Well. With a closer look it seems the original "Rerun" method is an awful hack that you write `NewHandle().Int(...).Common(...)` and use a boolean switch to control which one take effect.

We can just factor out the common logic as the test body and setup two parallel tests for it. And this is what @yedamao already did.

What if we add `t.Parallel()` at L353 for subtests?

Reply:
@xhebox @yedamao any updates?

Reply:
I can approve, the current solution works without problems. But it would be better to parallel sub tests.

Reply:
@xhebox@tisonkun ptal